### PR TITLE
Add error handling, refactor base controller

### PR DESCRIPTION
**Pull request by kozikkam**

### CODE REVIEW

File: backend/src/api/dietOrder/index.ts [hunk 1]
```diff
@@ -19,7 +20,7 @@ export const initDietOrderRoutes = (app: Application, prefix: string = "" ): voi
   const updateDietOrderController: IPutDietOrderController = container.get(DIET_ORDER_TYPES.IPutDietOrderController);
 
   const path = `${prefix}/dietOrder`;
-  app.post(path, postDietOrderController.process.bind(postDietOrderController));
-  app.get(path, getDietOrderController.process.bind(getDietOrderController));
-  app.put(`${path}/:id`, updateDietOrderController.process.bind(updateDietOrderController));
+  app.post(path, handleEndpointError(postDietOrderController.process.bind(postDietOrderController)));
```
Comment (maaaaagda) [inline backend/src/api/dietOrder/index.ts]: So I was testing your error handling for this endpoint and I believe that sending wrong request body should return sth else than 500 after hapi validation. Additionally validation error is printed to console
image

File: backend/src/core/errorHandler/handleEndpointError.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { NextFunction } from "connect";
+import { Request, Response } from "express";
+
+export const handleEndpointError: (handle: (req: Request, res: Response, next: NextFunction) => any) => any = 
+    (handle: (req: Request, res: Response, next: NextFunction) => any): any => {
```
[the remark this answers is not in the document]
Comment (kozikkam) [reply backend/src/core/errorHandler/handleEndpointError.ts]: It's not possible, unless I don't know about something. If you know of better solution, do tell :stuck_out_tongue:

[the remark this answers is not in the document]
Comment (kozikkam) [reply backend/src/core/errorHandler/handleEndpointError.ts]: Okay, I can adjust tslint rules so we don't have to add typings to arrow functions.

File: backend/src/api/dietOrder/index.ts (hunk 1 above)
Comment (kozikkam) [reply backend/src/api/dietOrder/index.ts]: It's good that it's printed to console, that's part of logging :stuck_out_tongue: Yeah perhaps error logging isn't done properly.

Fixed error handling.

**Pull request merged**